tests/src/Kernel/Validators/ValidatorEmptyCellProcessTest.php

```diff
@@ -0,0 +1,452 @@
+      [
+        2 => [
+          'case' => 'Empty value found in required column(s)',
+          'valid' => FALSE,
+          'failedItems' => [
+            'empty_indices' => [0, 2, 4],
+          ],
+        ],
+        6 => [
+          'case' => 'Empty value found in required column(s)',
+          'valid' => FALSE,
+          'failedItems' => [
+            'empty_indices' => [4, 5],
+          ],
+        ],
+      ],
+      $tokens,
+      $metadata,
+      [
+        'expected_message' => 'The following line number and column header combinations were empty, but a value is required.',
+        2 => [
+          'expected_columns' => 'Trait Name, Method Short Name, Unit',
+        ],
+        6 => [
+          'expected_columns' => 'Unit, Type',
+        ],
+      ],
+    ];
+
+    return $scenarios;
+  }
+
+  /**
+   * Tests the message processor method for the EmptyCell validator.
+   *
+   * @param array $failures
+   *   The failures array that gets passed to the process method. It contains
+   *   the following keys:
+   *   - The line number that triggered this failed validation status.
+   *     - 'case': a developer-focused string describing the case checked.
+   *     - 'valid': FALSE to indicate that validation failed.
+   *     - 'failedItems': an array of items that failed with the following keys.
+   *       - 'empty_indices': A list of column indices in the line which were
+   *         checked and found to be empty.
+   * @param array $tokens
+   *   An array of tokens to use for altering the messages that get displayed
+   *     to the user. The key is the token, (ex. 'project'), and the value is
+   *     the new value to be shown for that token.
+   * @param array $metadata
+   *   An array of additional metadata (or contextual information) needed by the
+   *   process method. Here, the following keys are expected:
+   *     - 'column_headers': This contains an array of headers. The index in
+   *       this array MUST match the position (starting with 0) of the column in
+   *       the input file.
+   *       Eg: 'column_headers' => [
+   *             '2' => 'Header 1', // Header of column #3
+   *             '4' => 'Header 2', // Header of column #5
+   *           ].
+   * @param array $expectations
+   *   An array of expectations that we want to find in the resulting rendered
+   *   output which has the following keys:
+   *   - 'expected_message': The message expected in the return value of the
+   *       process method for this scenario.
+   *   - 1+ arrays keyed by the line number in the input file that triggered
+   *     the failed validation status, further keyed by:
+   *     - 'expected_columns': A comma-separated list of column headers that
+   *       map to the expected columns with empty values.
+   *
+   * @dataProvider provideEmptyCellFailedCases
+   */
+  public function testProcessListWithDescribedTable(array $failures, array $tokens, array $metadata, array $expectations) {
```

Comment:
This will also need to updated in the docs and in the code 😅 
```suggestion
  public function testProcessListWithDescribedTable(array $validation_results, array $tokens, array $metadata, array $expectations) {
```

Reply:
It looks like this suggestion got missed! Remember to also update it in the doc